File: Dairyncia-dotnet/DTOs/CreateMilkCollectionDto.cs
using Dairyncia.Enums;
using System.ComponentModel.DataAnnotations;

public class CreateMilkCollectionDto
{
    [Required(ErrorMessage = "Farmer email is required")]
    [EmailAddress(ErrorMessage = "Invalid email format")]
    public string FarmerEmail { get; set; }

    [Required(ErrorMessage = "Milk type is required")]
    public MilkType MilkType { get; set; }

    [Required(ErrorMessage = "Milk shift is required")]
    public MilkShift MilkShift { get; set; }

    [Range(0.1, 1000, ErrorMessage = "Quantity must be between 0.1 and 1000 liters")]
    public decimal Quantity { get; set; }

    [Range(0, 10, ErrorMessage = "Fat percentage must be between 0 and 10")]
    public decimal FatPercentage { get; set; }

    [Range(0, 10, ErrorMessage = "SNF must be between 0 and 10")]
    public decimal SNF { get; set; }
}

File: Dairyncia-Spring/src/main/java/com/dairyncia/dto/CreateMilkCollectionDto.java
package com.dairyncia.dto;

import java.math.BigDecimal;

import com.dairyncia.enums.MilkShift;
import com.dairyncia.enums.MilkType;

import jakarta.validation.constraints.DecimalMax;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateMilkCollectionDto {

    @NotBlank(message = "Farmer email is required")
    @Email(message = "Invalid email format")
    private String farmerEmail;

    @NotNull(message = "Milk type is required")
    private MilkType milkType;

    @NotNull(message = "Milk shift is required")
    private MilkShift milkShift;

    @NotNull(message = "Quantity is required")
    @DecimalMin(value = "0.1", message = "Quantity must be between 0.1 and 1000 liters")
    @DecimalMax(value = "1000.0", message = "Quantity must be between 0.1 and 1000 liters")
    private BigDecimal quantity;

    @NotNull(message = "Fat percentage is required")
    @DecimalMin(value = "0.0", message = "Fat percentage must be between 0 and 10")
    @DecimalMax(value = "10.0", message = "Fat percentage must be between 0 and 10")
    private BigDecimal fatPercentage;

    @NotNull(message = "SNF is required")
    @DecimalMin(value = "0.0", message = "SNF must be between 0 and 10")
    @DecimalMax(value = "10.0", message = "SNF must be between 0 and 10")
    private BigDecimal snf;
}
